File: Python Implementation/Algorithms/Sorting/bubble_sort.py
def simple_bubble(arr):
    counter = 0

    for i in range(len(arr)):
        swap = False

        for j in range(len(arr)-i-1):
            counter += 1
            # j goes up to n-i because the last element will already be sorted
            # n-i-1 because j will be comparing the adjacent values so will get out of bounds without -1
            if arr[j] > arr[j+1]:
                # swap
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
                swap = True
        
        if swap == False:
            # if no swaps are made then the array is in order
            break
    
    print(counter)
    return arr

if __name__ == "__main__":
    x = [2, 7, 4, 1, 5, 3]
    print(simple_bubble(x))

File: C++ Implementation/Algorithms/Sorting/BubbleSort.cpp
void bubbleSort(int arr[], int n){
    for(int i=0; i<n; i++){
        bool swap = false;

        for(int j=0; j<n-i-1; j++){
            if(arr[j] > arr[j+1]){
                // swap
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;

                swap = true;
            }
        }

        if(!swap){
            break;
        }
    }
}

int main(){
    int n = 5;
    int arr[5] = {5, 1, 4, 2, 8};
    bubbleSort(arr, n);
    return 0;
}